core-bundle/tests/Security/Authentication/Provider/AuthenticationProviderTest.php

```diff
@@ -120,6 +121,8 @@ public function testLocksUserOnInvalidTwoFactorCode(int $initialAttempts, int $l
         $provider = $this->createTwoFactorProvider($authProvider, $userChecker);
         $hasException = false;
 
+        ClockMock::withClockMock(true);
```

Comment:
I think this also needs to be registered - either by calling `ClockMock::register()` or by adding the `@time-sensitive` annotation on the method/in the `phpunit.xml`. You can with `sleep(1000);` for instance - this should pass instantly.

Reply:
Just noticed, that we have already got some classes in the `phpunit.xml`:

https://github.com/contao/contao/blob/2ca2e45476e5a98aba8644093551a6576fa37261/phpunit.xml.dist#L65-L68

Reply:
There are whole directories. I'd rather register the `ClockMock` in the test, what do you think?

Reply:
Sounds sensible to me!

I'm personally in favor of the method annotation, but not sure what @leofeyer thinks about this?

Reply:
annotation is also fine for me :)

Reply:
It’s not about directories but about namespaces, ClockMock works by creating a function in the registered namespaces so that the classes in these namespaces use the moked functions instead of the native php functions.

We have to register the namespaces in the phpunit xml file because registering it in the tests can be too late (if the class was already loaded by a different test).

Reply:
Thanks for the clarification. I'll adjust it.

Reply:
Added to the list in b5486f939.